reactivesocket-core/src/test/java/io/reactivesocket/frame/VersionFlyweightTest.java

```diff
@@ -0,0 +1,21 @@
+package io.reactivesocket.frame;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class VersionFlyweightTest {
+    @Test
+    public void simple() {
+        int version = VersionFlyweight.encode(1, 0);
+        assertEquals(1, VersionFlyweight.major(version));
```

Comment:
Can you add a test that doesn't rely on the main code to both cases?

assertEquals(0x0100, version);